### ISSUE TITLE
[TW#14848] Incorrect Assertion in returnItemToRingbufDefault

### ISSUE THREAD

**Issue by HendrikSP** [espressif/esp-idf#920]
**Problem:**

The assertion in returnItemToRingbufDefault (line 340 in `components/freertos/ringbuf.c`)
```
configASSERT(data < rb->data+rb->size);
```
is wrong and needs to be
```
configASSERT(data <= rb->data+rb->size);
```

**Explanation:**

In `getItemFromRingbufDefault` the pointer is set to
```
ret=rb->read_ptr+sizeof(buf_entry_hdr_t)
```
If `hdr->len` is 0 then `rb->read_ptr` could be `rb->data+rb->size-sizeof(buf_entry_hdr_t)` and
the assertion  above will fail on returning the item.

If fixed the assertion locally and everything runs fine.

**Comment by Spritetm** [comment]
Gotcha. Interesting edge case, this. The fix is in our internal queue.

**Comment by igrr** [comment]
Fixed in 4da7ee1295172fb0f64d8cfa1711170ed782ca42.

**Comment by HendrikSP** [comment]
Thanks!
